compiler/util/files.cpp

```diff
@@ -181,12 +171,12 @@ const char* makeTempDir(const char* dirPrefix) {
   char* myuserid = strdup(userid);
   removeSpacesBackslashesFromString(myuserid);
 
-  const char* tmpDir = astr(tmpdirprefix, myuserid, mypidstr, tmpdirsuffix);
-  ensureDirExists(tmpDir, "making temporary directory");
+  const char* tmpDir = astr(tmpdirprefix, myuserid, tmpdirsuffix);
+  const char* dirRes = mkdtemp((char*) tmpDir);
```

Comment:
In an issue or chat there was some discussion about the portability of `mkdtemp` and needing different headers on different systems. What was the net result of that? Are we already including the right headers for other functionality so it just worked out?

Reply:
So far as I can tell, yes - the unistd header is already included in this file (and I checked it works on my Mac) and it built on Linux as well.  I haven't checked other configurations but those were the places I was worried about